<?xml version="1.0" encoding="UTF-8"?>

<launch>

  <arg name="model" default="$(find ros_practice)/urdf/arm_robot.urdf"/>
  <param name="robot_description" command="$(find xacro)/xacro $(arg model)"/>

  <include file="$(find gazebo_ros)/launch/empty_world.launch"/>

  <node name="spawn_urdf" pkg="gazebo_ros" type="spawn_model" args="-param robot_description -urdf -model arm_robot"/>

  <!-- Set parameters for "ros_controll" -->
  <rosparam>
    arm_robot: <!-- This must be the same of "ns" in following node -->
      arm1_arm2_joint_position_controller:
        type: effort_controllers/JointPositionController
        joint: joint_arm1_arm2
        pid: {p: 100.0, i: 0.01, d: 10.0}
      arm2_arm3_joint_position_controller:
        type: effort_controllers/JointPositionController
        joint: joint_arm2_arm3
        pid: {p: 100.0, i: 0.01, d: 10.0}
  </rosparam>

  <!-- Start ROS node of "/controller_spawner" -->
  <node name="controller_spawner1"
        pkg="controller_manager"
	type="spawner"
	respawn="false"
	output="screen"
	ns="/arm_robot"
	args="arm1_arm2_joint_position_controller"/>

  <node name="controller_spawner2"
	pkg="controller_manager"
	type="spawner"
	respawn="false"
	output="screen"
	ns="/arm_robot"
	args="arm2_arm3_joint_position_controller"/>
  
</launch>
